tests/helpers/perf_results_helper.rb

```diff
@@ -596,21 +622,52 @@ def percent_diff_string(result_a, result_b)
   #
   # @author Bill Claytor
   #
-  # @param [String] metrics_dir The 'puppet-metrics-collector' directory
+  # @param [String] metrics_dir_or_tar_file The 'puppet-metrics-collector' directory or tar file
   #
   # @return [void]
   #
   # @example
-  #   extract_puppet_metrics_collector_data(metrics_dir)
+  #   extract_puppet_metrics_collector_data(metrics_dir_or_tar_file)
   #
-  def extract_puppet_metrics_collector_data(metrics_dir)
-    raise "Directory not found: #{metrics_dir}" unless File.directory?(metrics_dir)
+  def extract_puppet_metrics_collector_data(metrics_dir_or_tar_file)
+    raise "File not found: #{metrics_dir_or_tar_file}" unless File.exist?(metrics_dir_or_tar_file)
+
+    if File.directory?(metrics_dir_or_tar_file)
+      puppet_metrics_dir = metrics_dir_or_tar_file
+    else
+      puts "Specified path is not a directory; verifying file..."
+      parent_dir = File.dirname(metrics_dir_or_tar_file)
+      extract_tarball(metrics_dir_or_tar_file)
+      puppet_metrics_dir = "#{parent_dir}/#{PUPPET_METRICS_COLLECTOR_DIR_NAME}"
+    end
 
-    puts "Extracting metrics data from: #{metrics_dir}"
+    puts "Extracting metrics data from: #{puppet_metrics_dir}"
     puts
 
     # TODO: process other service files
-    extract_puppetserver_metrics(metrics_dir)
+    extract_puppetserver_metrics(puppet_metrics_dir)
+  end
+
+  # Extracts the specified tarball to the specified directory
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] src The tarball path
+  # @param [String] dest The destination directory
+  #
+  # @return [void]
+  #
+  # @example
+  #   extract_tarball(src, dest)
+  #
+  def extract_tarball(src, dest = File.dirname(src))
+    options = ["file", "--brief", "--mime-type", src]
+    mime_type = IO.popen(options, in: :close, err: :close).read.chomp
+    error_msg = "Invalid mime type '#{mime_type}' for file: #{src}"
+    raise error_msg unless mime_type.include?("gzip")
```

Comment:
This method specifies that it expects a tar file, not necessarily a zgipped tar file. I would suggest that the call to Zlib get called _if_ the provided source is of type `gzip`, but it should also accept file types whose mime-type is `application/x-tar` as well.

Reply:
In this case extract_puppet_metrics_collector_data is only expected to handle the output of `collect_metrics_files.rb`. The separate `extract_tarball` method was 'extracted' from the calling method to streamline the code and facilitate testing when it was updated from calling a simple command to requiring `Zlib::GzipReader` and `Archive::Tar::Minitar`. It is not intended to be a generic tar file extraction utility that can handle multiple file types, so I've renamed the method and updated the specification to clarify that only gzipped tar files are accepted. This seems preferable to implementing additional functionality to handle both gzipped and non-gzipped tar files that is not expected to actually be used.